src/tests/sqlsmith/src/sql_gen/table_functions.rs

```diff
@@ -122,6 +123,207 @@ impl<R: Rng> SqlGenerator<'_, R> {
         }
     }
 
+    /// Generates `REGEXP_MATCH`.
+    /// `REGEXP_MATCH(input_string, pattern [, optional_flag ])`
+    fn gen_regexp_match(&mut self) -> (TableFactor, Table) {
+        let input_string = self.gen_random_string();
+        let is_match = self.flip_coin();
+        let pattern = if is_match {
+            self.gen_matched_regexp_pattern(&input_string, 3)
+        } else {
+            self.gen_noise_regexp_pattern(&input_string)
+        };
+        let flags = self.gen_regex_flags();
+
+        let input_expr = Expr::Value(Value::SingleQuotedString(input_string));
+        let pattern_expr = Expr::Value(Value::SingleQuotedString(pattern));
+        let flags_expr = Expr::Value(Value::SingleQuotedString(flags));
+
+        let table_name = self.gen_table_name_with_prefix("regexp_match");
+        let alias = create_table_alias(&table_name);
+        let relation = create_tvf(
+            "regexp_match",
+            alias,
+            create_args(vec![input_expr, pattern_expr, flags_expr]),
+            false,
+        );
+
+        let table = Table::new(table_name, vec![]);
+        (relation, table)
+    }
+
+    /// Recursively generates a regular expression pattern that matches the given input string.
+    /// The strategy includes:
+    /// - Escaping literals
+    /// - Replacing parts with character classes, alternations, or grouped sub-patterns
+    /// - Applying random quantifiers
+    fn gen_matched_regexp_pattern(&mut self, input: &str, depth: usize) -> String {
+        if depth == 0 || input.len() <= 1 {
+            return self.add_quantifier(regex::escape(input));
+        }
+
+        let chars: Vec<char> = input.chars().collect();
+        let len = chars.len();
+        let (start, end) = self.rand_segment(len);
+        let (prefix, mid, suffix) = (&input[..start], &input[start..end], &input[end..]);
+
+        let mid_transformed = match self.rng.random_range(0..=4) {
+            0 => self.gen_char_class(mid.chars().next().unwrap_or('a')),
```

Comment:
Is it possible that chars are empty? If yes, we should not unwrap to an `'a'`. Otherwise it can be unmatched by the regex.